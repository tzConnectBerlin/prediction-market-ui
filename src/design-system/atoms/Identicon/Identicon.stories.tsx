import * as React from 'react';
// also exported from '@storybook/react' if you can deal with breaking changes in 6.1
import { Story, Meta } from '@storybook/react/types-6-0';

import { Identicon, IdenticonProps } from './Identicon';

export default {
  title: 'Atom/Identicon',
  component: Identicon,
} as Meta;

const Template: Story<IdenticonProps> = (args) => <Identicon {...args} />;

export const IdenticonWithSeed = Template.bind({});
IdenticonWithSeed.args = {
  seed: 'QmYgtfMBZo3ajW5rmUesVfHSJu5nT6fT3cRcvr2fpfbzo3',
};

export const IdenticonWithURL = Template.bind({});
IdenticonWithURL.args = {
  url: 'https://w.wallhaven.cc/full/vg/wallhaven-vg7lv3.jpg',
};

export const IdenticonWithBackground = Template.bind({});
IdenticonWithBackground.args = {
  type: 'tzKtCat',
  seed: 'tz1VSUr8wwNhLAzempoch5d6hLRiTh8Cjcjb',
};

export const IdenticonMaxSize = Template.bind({});
IdenticonMaxSize.args = {
  url: 'https://w.wallhaven.cc/full/rd/wallhaven-rd3pjw.jpg',
  iconSize: 'max',
  hasBackground: false,
};
