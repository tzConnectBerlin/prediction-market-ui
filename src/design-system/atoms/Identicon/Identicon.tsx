import * as React from 'react';
import { BlockiesOptions, create } from 'blockies-ts';
import { Avatar, AvatarProps } from '@material-ui/core';
import styled from '@emotion/styled';
import { lightTheme as theme } from '../../../styles/theme';

export const StyledAvatar = styled(Avatar)`
  &.hasBackground {
    background-color: rgba(29, 34, 39, 0.04);
  }

  &.xs {
    width: ${theme.spacing(3)};
    height: ${theme.spacing(3)};
  }

  &.sm {
    width: ${theme.spacing(4)};
    height: ${theme.spacing(4)};
  }

  &.md {
    width: ${theme.spacing(5)};
    height: ${theme.spacing(5)};
  }

  &.lg {
    width: ${theme.spacing(6)};
    height: ${theme.spacing(6)};
  }

  &.xl {
    width: ${theme.spacing(8)};
    height: ${theme.spacing(8)};
  }

  &.xxl {
    width: ${theme.spacing(9.625)};
    height: ${theme.spacing(9.625)};
  }

  &.max {
    width: ${theme.spacing(18)};
    height: ${theme.spacing(18)};
  }
`;

export interface IdenticonProps extends Omit<Partial<BlockiesOptions>, 'seed'> {
  type?: 'blockies' | 'tzKtCat';
  seed?: string;
  url?: string;
  variant?: AvatarProps['variant'];
  alt?: string;
  iconSize?: 'xs' | 'sm' | 'md' | 'lg' | 'xl' | 'xxl' | 'max';
  hasBackground?: boolean;
  onClick?: (event?: React.MouseEvent<any>) => void | Promise<void>;
}

// TODO: redo with forwardRef
export const Identicon: React.FC<IdenticonProps> = ({
  type = 'blockies',
  seed,
  variant = 'circular',
  alt,
  url,
  iconSize = 'xl',
  hasBackground = true,
  onClick,
  ...rest
}) => {
  const data =
    url ||
    // eslint-disable-next-line no-nested-ternary
    (type === 'tzKtCat' && seed
      ? `https://services.tzkt.io/v1/avatars2/${seed}`
      : type === 'blockies' && seed
      ? create({ seed, ...rest }).toDataURL()
      : undefined);
  return (
    <StyledAvatar
      variant={variant}
      alt={alt}
      src={data}
      onClick={onClick}
      className={[iconSize || undefined, hasBackground ? 'hasBackground' : undefined].join(' ')}
    />
  );
};
