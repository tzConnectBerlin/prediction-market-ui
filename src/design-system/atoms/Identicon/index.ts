export type { IdenticonProps } from './Identicon';
export { Identicon, Identicon as default } from './Identicon';
