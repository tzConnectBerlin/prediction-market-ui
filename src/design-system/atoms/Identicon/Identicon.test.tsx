import renderer from 'react-test-renderer';
import { Identicon, IdenticonProps } from './Identicon';

const defaultProps: IdenticonProps = {
  type: 'tzKtCat',
  seed: 'tz1VSUr8wwNhLAzempoch5d6hLRiTh8Cjcjb',
  url: 'https://w.wallhaven.cc/full/vg/wallhaven-vg7lv3.jpg',
};
describe('Snapshot testing Identicon Component', () => {
  it('renders correctly without props', () => {
    const icon = renderer.create(<Identicon />).toJSON();
    expect(icon).toMatchSnapshot();
  });

  it('renders correctly with Url', () => {
    const icon = renderer.create(<Identicon url={defaultProps.url} />).toJSON();
    expect(icon).toMatchSnapshot();
  });

  it('renders correctly with seed', () => {
    const icon = renderer.create(<Identicon seed={defaultProps.seed} />).toJSON();
    expect(icon).toMatchSnapshot();
  });

  it('renders correctly with tzKtCat type with Background', () => {
    const icon = renderer
      .create(<Identicon seed={defaultProps.seed} type={defaultProps.type} />)
      .toJSON();
    expect(icon).toMatchSnapshot();
  });

  it('renders correctly with tzKtCat type without Background', () => {
    const icon = renderer
      .create(<Identicon seed={defaultProps.seed} type={defaultProps.type} hasBackground={false} />)
      .toJSON();
    expect(icon).toMatchSnapshot();
  });
});
