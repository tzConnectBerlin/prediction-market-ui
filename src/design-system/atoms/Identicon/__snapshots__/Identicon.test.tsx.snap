// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing Identicon Component renders correctly with Url 1`] = `
.emotion-0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 40px;
  height: 40px;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-size: 1.25rem;
  line-height: 1;
  border-radius: 50%;
  overflow: hidden;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.emotion-0.hasBackground {
  background-color: rgba(29, 34, 39, 0.04);
}

.emotion-0.xs {
  width: 24px;
  height: 24px;
}

.emotion-0.sm {
  width: 32px;
  height: 32px;
}

.emotion-0.md {
  width: 40px;
  height: 40px;
}

.emotion-0.lg {
  width: 48px;
  height: 48px;
}

.emotion-0.xl {
  width: 64px;
  height: 64px;
}

.emotion-0.xxl {
  width: 77px;
  height: 77px;
}

.emotion-0.max {
  width: 144px;
  height: 144px;
}

.emotion-1 {
  width: 100%;
  height: 100%;
  text-align: center;
  object-fit: cover;
  color: transparent;
  text-indent: 10000px;
}

<div
  className="MuiAvatar-root MuiAvatar-circular xl hasBackground emotion-0"
>
  <img
    className="MuiAvatar-img emotion-1"
    src="https://w.wallhaven.cc/full/vg/wallhaven-vg7lv3.jpg"
  />
</div>
`;

exports[`Snapshot testing Identicon Component renders correctly with seed 1`] = `
.emotion-0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 40px;
  height: 40px;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-size: 1.25rem;
  line-height: 1;
  border-radius: 50%;
  overflow: hidden;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.emotion-0.hasBackground {
  background-color: rgba(29, 34, 39, 0.04);
}

.emotion-0.xs {
  width: 24px;
  height: 24px;
}

.emotion-0.sm {
  width: 32px;
  height: 32px;
}

.emotion-0.md {
  width: 40px;
  height: 40px;
}

.emotion-0.lg {
  width: 48px;
  height: 48px;
}

.emotion-0.xl {
  width: 64px;
  height: 64px;
}

.emotion-0.xxl {
  width: 77px;
  height: 77px;
}

.emotion-0.max {
  width: 144px;
  height: 144px;
}

.emotion-1 {
  width: 100%;
  height: 100%;
  text-align: center;
  object-fit: cover;
  color: transparent;
  text-indent: 10000px;
}

<div
  className="MuiAvatar-root MuiAvatar-circular xl hasBackground emotion-0"
>
  <img
    className="MuiAvatar-img emotion-1"
    src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAABmJLR0QA/wD/AP+gvaeTAAAArElEQVRYhWPcfZ39PwMS4F2mxkBL8DnqFgqfiaa2EQFGHcCCHufocYQu36TzBa+BdVd4SDJvwENg1AEshBSgx2EdgXICXT0hMOAhMOoAjDRAqC4gFMek1iUDHgKjDmAhFKf9l6RQ+IUEDESvKwr1nuFVP+AhMOoAgnUBOiCYzwm0F9DBgIfAqAMYT9TqovQLCLXhSAWjbcJB7wAWjLKbgAZK2wPo9g14CIw6AAC/UDGGEGUY5QAAAABJRU5ErkJggg=="
  />
</div>
`;

exports[`Snapshot testing Identicon Component renders correctly with tzKtCat type with Background 1`] = `
.emotion-0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 40px;
  height: 40px;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-size: 1.25rem;
  line-height: 1;
  border-radius: 50%;
  overflow: hidden;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.emotion-0.hasBackground {
  background-color: rgba(29, 34, 39, 0.04);
}

.emotion-0.xs {
  width: 24px;
  height: 24px;
}

.emotion-0.sm {
  width: 32px;
  height: 32px;
}

.emotion-0.md {
  width: 40px;
  height: 40px;
}

.emotion-0.lg {
  width: 48px;
  height: 48px;
}

.emotion-0.xl {
  width: 64px;
  height: 64px;
}

.emotion-0.xxl {
  width: 77px;
  height: 77px;
}

.emotion-0.max {
  width: 144px;
  height: 144px;
}

.emotion-1 {
  width: 100%;
  height: 100%;
  text-align: center;
  object-fit: cover;
  color: transparent;
  text-indent: 10000px;
}

<div
  className="MuiAvatar-root MuiAvatar-circular xl hasBackground emotion-0"
>
  <img
    className="MuiAvatar-img emotion-1"
    src="https://services.tzkt.io/v1/avatars2/tz1VSUr8wwNhLAzempoch5d6hLRiTh8Cjcjb"
  />
</div>
`;

exports[`Snapshot testing Identicon Component renders correctly with tzKtCat type without Background 1`] = `
.emotion-0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 40px;
  height: 40px;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-size: 1.25rem;
  line-height: 1;
  border-radius: 50%;
  overflow: hidden;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.emotion-0.hasBackground {
  background-color: rgba(29, 34, 39, 0.04);
}

.emotion-0.xs {
  width: 24px;
  height: 24px;
}

.emotion-0.sm {
  width: 32px;
  height: 32px;
}

.emotion-0.md {
  width: 40px;
  height: 40px;
}

.emotion-0.lg {
  width: 48px;
  height: 48px;
}

.emotion-0.xl {
  width: 64px;
  height: 64px;
}

.emotion-0.xxl {
  width: 77px;
  height: 77px;
}

.emotion-0.max {
  width: 144px;
  height: 144px;
}

.emotion-1 {
  width: 100%;
  height: 100%;
  text-align: center;
  object-fit: cover;
  color: transparent;
  text-indent: 10000px;
}

<div
  className="MuiAvatar-root MuiAvatar-circular xl  emotion-0"
>
  <img
    className="MuiAvatar-img emotion-1"
    src="https://services.tzkt.io/v1/avatars2/tz1VSUr8wwNhLAzempoch5d6hLRiTh8Cjcjb"
  />
</div>
`;

exports[`Snapshot testing Identicon Component renders correctly without props 1`] = `
.emotion-0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 40px;
  height: 40px;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-size: 1.25rem;
  line-height: 1;
  border-radius: 50%;
  overflow: hidden;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  color: #fff;
  background-color: #bdbdbd;
}

.emotion-0.hasBackground {
  background-color: rgba(29, 34, 39, 0.04);
}

.emotion-0.xs {
  width: 24px;
  height: 24px;
}

.emotion-0.sm {
  width: 32px;
  height: 32px;
}

.emotion-0.md {
  width: 40px;
  height: 40px;
}

.emotion-0.lg {
  width: 48px;
  height: 48px;
}

.emotion-0.xl {
  width: 64px;
  height: 64px;
}

.emotion-0.xxl {
  width: 77px;
  height: 77px;
}

.emotion-0.max {
  width: 144px;
  height: 144px;
}

.emotion-1 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  width: 75%;
  height: 75%;
}

<div
  className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault xl hasBackground emotion-0"
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback emotion-1"
    data-testid="PersonIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
    />
  </svg>
</div>
`;
